api/score.ts

```diff
@@ -0,0 +1,59 @@
+import { VercelRequest, VercelResponse } from '@vercel/node';
+import OpenAI from 'openai';
+import { z } from 'zod';
+
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+const secretKey = process.env.SECRET_KEY || '';
+
+// Define the schema for the request body
+const RequestSchema = z.object({
+  content: z.record(z.unknown()),
+});
+
+export default async function handler(request: VercelRequest, response: VercelResponse) {
+  try {
+    // Validate the authorization header
+    const authHeader = request.headers?.['authorization'] ?? '';
+    if (authHeader !== `Bearer ${secretKey}`) {
+      response.statusCode = 401;
+      response.setHeader('Content-Type', 'application/json');
+      response.end(JSON.stringify({ error: 'Unauthorized' }));
+      return;
+    }
+
+    // Validate the request body
+    const validationResult = RequestSchema.safeParse(request.body);
+    if (!validationResult.success) {
+      response.statusCode = 400;
```

Comment:
```suggestion
      response.statusCode = 422;
```

This is the correct response code for validation errors